website_sale_product_contract_gift/models/sale_order.py

```diff
@@ -94,13 +94,20 @@ def action_confirm(self):
             if existing_partners:
                 contract_partner = existing_partners[0]
             else:
-                contract_partner = order.partner_shipping_id.copy(
+                contract_partner = order.partner_shipping_id.create(
```

Comment:
i know that this was already the case before this change, but i don’t understand: why is a new partner created with the same values instead of using the `order.partner_shipping_id` partner directly? if there is a good reason, a comment here would help.

regardless, i think this is more easily understandable:
```suggestion
                contract_partner = self.env["res.partner"].create(
```